server/index.js

```diff
@@ -23,14 +23,60 @@ app.use(bodyParser.urlencoded({ extended: false }));
 
 
 app.get('/carousels/:id', (req, res) => {
-  Carousels.find({id: req.params.id}, (err, results) => {
-    if (err) {
-      return console.log('error getting from db: ', err)
-    }
-    res.json(results)
-  })
-})
+  console.log('req.params.id: ', req.params.id);
+  const { id } = req.params;
+  // res.status(200).json('res for get request');
+  model.findOne(id)
+    .then((data) => {
+      res.status(200).json(data);
+    })
+    .catch((err) => {
+      console.log('')
+    });
+});
+
+app.post('/carousels', (req, res) => {
+  let data = req.body;
+  if (data._id) {
+    res.status(403).send('cannot post to database with existing _id');
+  }
+  return model.insertOne(data)
+    .then((response) => {
+      console.log('response from insertOne: ', response);
+      res.status(200).json(response);
+    })
+    .catch((err) => {
+      console.log('error in app.post() endpoing: ', err);
+      res.status(500).send(err);
+    });
+});
 
+app.put('/carousels/:id', (req, res) => {
+  const { id } = req.params;
+  const data = req.body;
+  model.updateCarousel(id, data)
+    .then((response) => {
+      console.log('response from updateCarousel: ', response);
+      res.status(200).json(response);
+    })
+    .catch((err) => {
+      console.log('error inside endpoint app.put(): ', err);
+      res.status(500).json(err);
+    });
+});
+
+app.delete('/carousels/:id', (req, res) => {
+  const { id } = req.params;
+  model.deleteCarousel(id)
+    .then((response) => {
+      console.log('response in endpoint app.delete(): ', response);
+      res.status(200).send(response);
+    })
+    .catch((err) => {
+      console.log('error in endpoint app.delete(): ', err);
+      res.status(500).json(err);
+    });
+});
```

Comment:
These look great as well.

Once this is ready for deployment, I like to use a tool like [strip-debug](https://www.npmjs.com/package/strip-debug) to remove all of the console logs so I don't have to go back and manually take them out.